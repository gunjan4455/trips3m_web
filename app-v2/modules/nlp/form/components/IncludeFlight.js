import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import styles from './IncludeFlight.cm.scss';

class IncludeFlight extends PureComponent {

  handleChange = () => {
    const { keyValue, value,  onChange } = this.props;
    onChange(keyValue, !value);
  };

  render() {
    return (
      <div className={`at_includeFlight p24 flex mb64 spaceBetween ${styles.bottomSpacing}`}>
        <p className="fw9">Flights</p>
        <div className={styles.toggleBtn}>
          <label className={styles.switch}>
            <input name="flights" type="checkbox" checked={this.props.value} onChange={this.handleChange}/>
              <span className={styles.slider}/>
          </label>
        </div>
      </div>
    );
  }
}

IncludeFlight.propTypes = {
  value: PropTypes.bool,
  keyValue: PropTypes.string.isRequired,
  onChange: PropTypes.func,
};

IncludeFlight.defaultProps = {
  value: false,
  onChange: () => {},
};


export default IncludeFlight;
