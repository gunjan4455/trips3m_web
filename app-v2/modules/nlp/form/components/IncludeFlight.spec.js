import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import IncludeFlight from './IncludeFlight';

configure({ adapter: new Adapter() });

const createShallowWrapper = (props) => {
  return shallow(<IncludeFlight {...props} />);
};

describe('Render Include Flight Component ', () => {

  it('should match with the previous snapshot', () => {
    const wrapper = createShallowWrapper();
    expect(FlightComponent).toMatchSnapshot();
  });

  it('should toggle state when input is changed ', () => {
    const wrapper = createShallowWrapper();
    const inputElement = wrapper.find('input');
    const prevValue = wrapper.state().value;
    inputElement.simulate('change');
    expect(wrapper.state().value).toEqual(!prevValue);
  });

});
