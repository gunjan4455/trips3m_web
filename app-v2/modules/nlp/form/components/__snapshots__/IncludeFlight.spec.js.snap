// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Include Flight Component  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IncludeFlight />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="fw9"
        >
          Flights
        </p>,
        <div
          className="toggleBtn"
        >
          <label
            className="switch"
          >
            <input
              name="flights"
              onChange={[Function]}
              type="checkbox"
              value={true}
            />
            <span
              className="slider"
            />
          </label>
        </div>,
      ],
      "className": "at_includeFlight p24 flex mb64 spaceBetween bottomSpacing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Flights",
          "className": "fw9",
        },
        "ref": null,
        "rendered": "Flights",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="switch"
          >
            <input
              name="flights"
              onChange={[Function]}
              type="checkbox"
              value={true}
            />
            <span
              className="slider"
            />
          </label>,
          "className": "toggleBtn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="flights"
                onChange={[Function]}
                type="checkbox"
                value={true}
              />,
              <span
                className="slider"
              />,
            ],
            "className": "switch",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "flights",
                "onChange": [Function],
                "type": "checkbox",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "slider",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="fw9"
          >
            Flights
          </p>,
          <div
            className="toggleBtn"
          >
            <label
              className="switch"
            >
              <input
                name="flights"
                onChange={[Function]}
                type="checkbox"
                value={true}
              />
              <span
                className="slider"
              />
            </label>
          </div>,
        ],
        "className": "at_includeFlight p24 flex mb64 spaceBetween bottomSpacing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Flights",
            "className": "fw9",
          },
          "ref": null,
          "rendered": "Flights",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="switch"
            >
              <input
                name="flights"
                onChange={[Function]}
                type="checkbox"
                value={true}
              />
              <span
                className="slider"
              />
            </label>,
            "className": "toggleBtn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  name="flights"
                  onChange={[Function]}
                  type="checkbox"
                  value={true}
                />,
                <span
                  className="slider"
                />,
              ],
              "className": "switch",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "flights",
                  "onChange": [Function],
                  "type": "checkbox",
                  "value": true,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "slider",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
